Microsoft.Toolkit.Uwp.UI.Controls/FlexPanel/FlexPanel.cs

```diff
@@ -0,0 +1,1613 @@
+            item.Basis = GetBasis(view);
+            item.AlignSelf = GetAlignSelf(view);
+
+            var margin = (Thickness)view.GetValue(MarginProperty);
+            item.MarginLeft = (double)margin.Left;
+            item.MarginTop = (double)margin.Top;
+            item.MarginRight = (double)margin.Right;
+            item.MarginBottom = (double)margin.Bottom;
+
+            var width = view.Width;
+            item.Width = width <= 0 ? double.NaN : (double)width;
+
+            var height = view.Height;
+            item.Height = height <= 0 ? double.NaN : (double)height;
+
+            item.IsVisible = (Visibility)view.GetValue(VisibilityProperty) == Visibility.Visible;
+        }
+
+        /// <summary>
+        /// UWP defined method
+        /// </summary>
+        /// <param name="finalSize">Size allocated to this element</param>
+        /// <returns>Size consumed by this element</returns>
+        protected override Size ArrangeOverride(Size finalSize)
+        {
+            var width = finalSize.Width;
+            var height = finalSize.Height;
+
+            Layout(width, height);
+
+            foreach (var child in Children)
+            {
+                if (GetFlexItem(child) is FlexItem item)
+                {
+                    var frame = item.GetFrame();
+                    if (double.IsNaN(frame.X)
+                        || double.IsNaN(frame.Y)
+                        || double.IsNaN(frame.Width)
+                        || double.IsNaN(frame.Height))
+                    {
+                        throw new Exception("something is deeply wrong");
+                    }
+
+                    child.Arrange(frame);
+                }
+            }
+
+            return finalSize;
+        }
+
+        private bool _measuring;
+
+        /// <summary>
+        /// UWP defined
+        /// </summary>
+        /// <param name="availableSize">how much space is available to this element</param>
+        /// <returns>size desired to render this element</returns>
+        protected override Size MeasureOverride(Size availableSize)
+        {
+            var widthConstraint = availableSize.Width;
+            var heightConstraint = availableSize.Height;
+
+            if (_root == null)
+            {
+                return new Size(widthConstraint, heightConstraint);
+            }
+
+            _measuring = true;
+
+            // 1. Keep track of missing layout items
+            var deleteCandidates = _root.ToList();
+
+            // 2. Set Shrink to 0, set align-self to start (to avoid stretching)
+            //    Set Image.Aspect to Fill to get the value we expect in measuring
+            foreach (var child in Children)
+            {
+                if (GetFlexItem(child) is FlexItem item && item.Parent != null)
+                {
+                    deleteCandidates.Remove(item);
+                }
+                else if (child is FrameworkElement frameworkElement)
```

Comment:
Why does this need to be cast to a FrameworkElement? The method `AddChild` only uses it as a UIElement